TriblerGUI/widgets/settingspage.py

```diff
@@ -192,11 +192,17 @@ def save_settings(self):
             max_conn_download = -1
         settings_data['libtorrent']['max_connections_download'] = max_conn_download
 
-        if self.window().upload_rate_limit_input.text():
-            settings_data['libtorrent']['max_upload_rate'] = int(self.window().upload_rate_limit_input.text()) * 1024
-        if self.window().download_rate_limit_input.text():
-            settings_data['libtorrent']['max_download_rate'] = int(self.window().download_rate_limit_input.text()) \
-                                                               * 1024
+        try:
+            if self.window().upload_rate_limit_input.text():
+                settings_data['libtorrent']['max_upload_rate'] = int(self.window().upload_rate_limit_input.text()) \
+                                                                 * 1024
+            if self.window().download_rate_limit_input.text():
+                settings_data['libtorrent']['max_download_rate'] = int(self.window().download_rate_limit_input.text()) \
+                                                                   * 1024
+        except ValueError:
+            ConfirmationDialog.show_error(self.window(), "Invalid number for maximum upload/download",
```

Comment:
You can give the user a bit more information here, something like:

```python
def get_illegal_characters(input)
    return reduce(lambda x, y: x + ('' if y.isdigit() else y), input, "")

print "Found the following illegal characters", get_illegal_characters(input)
```

Reply:
I've clarified the error a bit in such a way that the user is informed that a whole number is required. I think this error message should just be a small amount of text.